File: pv3-monitor/backend/app/schemas/alarm.py
from datetime import datetime

from pydantic import BaseModel


class AlarmResponse(BaseModel):
    """Schema for alarm response."""

    id: int
    device_id: str
    alarm_name: str
    is_active: bool
    updated_at: datetime

    class Config:
        from_attributes = True


class AlarmEventResponse(BaseModel):
    """Schema for alarm event response."""

    id: int
    device_id: str
    timestamp: datetime
    alarm_name: str
    event_type: str

    class Config:
        from_attributes = True


class AlarmStatus(BaseModel):
    """Schema for current alarm status overview."""

    device_id: str
    active_count: int
    active_alarms: list[str]
    all_alarms: dict[str, bool]
    updated_at: datetime

File: pv3-monitor/frontend/src/types/alarm.ts
export interface AlarmStatus {
  device_id: string
  active_count: number
  active_alarms: string[]
  all_alarms: Record<string, boolean>
  updated_at: string
}

export interface AlarmEvent {
  id: number
  device_id: string
  timestamp: string
  alarm_name: string
  event_type: 'triggered' | 'cleared'
}

